File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Cdm/CdmAttributeContextReference.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace Microsoft.CommonDataModel.ObjectModel.Cdm
{
    using Microsoft.CommonDataModel.ObjectModel.Enums;
    using Microsoft.CommonDataModel.ObjectModel.Utilities;
    using System;
    using System.Collections.Generic;

    public class CdmAttributeContextReference : CdmObjectReferenceBase
    {
        /// <summary>
        /// Constructs a CdmAttributeContextReference.
        /// </summary>
        /// <param name="ctx">The context.</param>
        /// <param name="name">The name.</param>
        public CdmAttributeContextReference(CdmCorpusContext ctx, string name)
            : base(ctx, name, true)
        {
            this.ObjectType = CdmObjectType.AttributeContextRef;
        }

        internal override CdmObjectReferenceBase CopyRefObject(ResolveOptions resOpt, dynamic refTo, bool simpleReference, CdmObjectReferenceBase host = null)
        {
            if (host == null)
                return new CdmAttributeContextReference(this.Ctx, refTo);
            else
                return host.CopyToHost(this.Ctx, refTo, simpleReference);
        }

        [Obsolete("InstanceFromData is deprecated. Please use the Persistence Layer instead.")]
        public static CdmAttributeContextReference InstanceFromData(CdmCorpusContext ctx, dynamic obj)
        {
            return CdmObjectBase.InstanceFromData<CdmAttributeContextReference, dynamic>(ctx, obj);
        }

        [Obsolete("CopyData is deprecated. Please use the Persistence Layer instead.")]
        public override dynamic CopyData(ResolveOptions resOpt, CopyOptions options)
        {
            return CdmObjectBase.CopyData<CdmAttributeContextReference>(this, resOpt, options);
        }

        [Obsolete]
        public override CdmObjectType GetObjectType()
        {
            return CdmObjectType.AttributeContextRef;
        }

        internal override bool VisitRef(string pathFrom, VisitCallback preChildren, VisitCallback postChildren)
        {
            return false;
        }
    }
}

File: objectModel/TypeScript/Cdm/CdmAttributeContextReference.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import {
    CdmCorpusContext,
    CdmObjectReferenceBase,
    cdmObjectType,
    resolveOptions,
    VisitCallback
} from '../internal';

export class CdmAttributeContextReference extends CdmObjectReferenceBase {
    public static get objectType(): cdmObjectType {
        return cdmObjectType.attributeContextRef;
    }

    constructor(ctx: CdmCorpusContext, name: string) {
        super(ctx, name, true);
        this.objectType = cdmObjectType.attributeContextRef;
    }

    public getObjectType(): cdmObjectType {
        return cdmObjectType.attributeContextRef;
    }

    /**
     * @internal
     */
    public copyRefObject(resOpt: resolveOptions, refTo: string, simpleReference: boolean, host?: CdmObjectReferenceBase): CdmObjectReferenceBase {
        if (!host) {
            return new CdmAttributeContextReference(this.ctx, refTo);
        } else {
            return host.copyToHost(this.ctx, refTo, simpleReference);
        }
    }

    /**
     * @internal
     */
    public visitRef(pathFrom: string, preChildren: VisitCallback, postChildren: VisitCallback): boolean {
        return false;
    }
}
